modules/dhcp/providers/server/native_ms.rb

```diff
@@ -92,30 +93,50 @@ def loadSubnetData subnet
       cmd = "scope #{subnet.network} show reservedip"
       msg = "Enumerated hosts on #{subnet.network}"
 
-      # Extract the data
+      netsh_tmp = Tempfile.new('smartproxydhcp')
+      netsh_tmp.write("dhcp server\n")
+      all_entries = Hash.new()
+      # Extract the data and build a scriptfile
       execute(cmd, msg).each do |line|
         #     172.29.216.6      -    00-a0-e7-21-41-00-
         if line =~ /^\s+([\w\.]+)\s+-\s+([-a-f\d]+)/
           ip  = $1
           next unless subnet.include?(ip)
           mac = $2.gsub(/-/,":").match(/^(.*?).$/)[1]
-          begin
-            opts = {:subnet => subnet, :ip => ip, :mac => mac}
-            opts.merge!(loadRecordOptions(opts))
-            logger.debug opts.inspect
-            if opts.include? :hostname
-              Proxy::DHCP::Reservation.new opts.merge(:deleteable => true)
-            else
-              # this is not a lease, rather reservation
-              # but we require option 12(hostname) to be defined for our leases
-              # workaround until #1172 is resolved.
-              Proxy::DHCP::Lease.new opts
-            end
-          rescue Exception => e
-            logger.warn "Skipped #{line} - #{e}"
-          end
+          opts = {:subnet => subnet, :ip => ip, :mac => mac}
+          netsh_tmp.write("scope #{opts[:subnet].network} Show ReservedOptionValue #{ip}\n")
+          all_entries[ip] = opts
         end
       end
+      netsh_tmp.close
+      # now run the script
+      script_output = execute_with_script(netsh_tmp.path,"Running NETSH in script mode")
+      # remove the tmpfile
+      netsh_tmp.unlink
+      # convert the output into an IP keyed hashtable
+      processed = parse_script_output_to_hash(script_output)
+      # now run through the single command parser
+      all_entries.each do |ip,existing_options|
+        query_result = nil
+        new_options = nil
+        query_result = processed[ip]
+        new_options = parse_options(query_result)
+        logger.debug "    query_result #{query_result.inspect}"
+        logger.debug "     new_options #{new_options.inspect}"
+        logger.debug "existing_options #{existing_options.inspect}"
+        existing_options.merge!(new_options)
+        logger.debug "merged_options #{existing_options.inspect}"
+        if existing_options.include? :hostname
+          logger.info "New reservation #{existing_options.inspect}"
+          Proxy::DHCP::Reservation.new existing_options.merge({:deleteable => true})
+        else
+          # this is not a lease, rather reservation
+          # but we require option 12(hostname) to be defined for our leases
+          # workaround until #1172 is resolved.
+          logger.info "New Lease #{existing_options.inspect}"
+          Proxy::DHCP::Lease.new existing_options
+         end
+      end
```

Comment:
You can extract lines 120-138 into a methods, which will help with testing. Can you add a test for this please?